aten/src/ATen/native/mps/operations/Sorting.mm

```diff
@@ -0,0 +1,75 @@
+#include <ATen/ATen.h>
+#include <ATen/Tensor.h>
+#include <ATen/Utils.h>
+#include <ATen/mps/MPSStream.h>
+#include <ATen/native/mps/OperationUtils.h>
+#include <torch/library.h>
+#include <c10/util/Optional.h>
+
+
+namespace at {
+namespace native {
+
+
+Tensor argsort_stable(const Tensor & self, bool stable, int64_t dim, bool descending) {
+
+  using namespace mps;
+  MPSStream* stream = getCurrentMPSStream();
+  Tensor result = std::get<1>(at::sort(self, stable, dim, descending));
+
+  struct CachedGraph : public MPSCachedGraph
+  {
+    CachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {}
+    MPSGraphTensor* inputTensor_ = nil;
+    MPSGraphTensor* outputTensor_ = nil;
+  };
+
+  MPSGraphCache* cache_ = MPSGraphCache::getInstance();
+
+  @autoreleasepool {
+    string key = "argsort_stable_mps:" + getTensorsStringKey({self});
+
+    CachedGraph* cachedGraph = static_cast<CachedGraph *>(cache_->LookUp(key));
+    if(!cachedGraph) {
+      MPSCachedGraph *tmpCachedGraph = cache_->CreateCachedGraph(key, ^ MPSCachedGraph * () {
+
+        CachedGraph *newCachedGraph = nil;
+
+        @autoreleasepool {
+          MPSGraph* mpsGraph = make_mps_graph();
+          newCachedGraph = new CachedGraph(mpsGraph);
+
+          MPSGraphTensor* inputTensor = mpsGraphRankedPlaceHolder(mpsGraph, self);
+
+          MPSGraphTensor* outputTensor = [mpsGraph argSortWithTensor:inputTensor
```

Comment:
So this API is available on OS Ventura onwards only. So we should add an if-check and pragma like in this [PR](https://github.com/pytorch/pytorch/pull/88319/files#diff-9b23535bcb52b7ec6b73bb2b078239abcd6a55a7d238851efffa09024c81db51R11)

Reply:
@kulinseth 
Thanks for your comment! I want to clarify what I should do because I'm not sure...

Do I have to add similar parts like these? If so, in which line should I put if check?

https://github.com/pytorch/pytorch/blob/657f2e12f0e212b2f4afd89ab2c824c409dcc951/aten/src/ATen/native/mps/operations/UnaryOps.mm#L10-L15
https://github.com/pytorch/pytorch/blob/657f2e12f0e212b2f4afd89ab2c824c409dcc951/aten/src/ATen/native/mps/operations/UnaryOps.mm#L274-L277
https://github.com/pytorch/pytorch/blob/657f2e12f0e212b2f4afd89ab2c824c409dcc951/aten/src/ATen/native/mps/operations/UnaryOps.mm#L286-L291

Obviously, I will need to change `Cumsum` to `Argsort`, `cumulativeSumWithTensor` to `argSortWithTensor` etc

Reply:
@kulinseth 
Please check my comment above when you have time 🙏 I'm not sure yet how to implement what you said

Reply:
@shogohida Have a look at my [Merged PR ](https://github.com/pytorch/pytorch/pull/88807)
Still don't understand, tag me. I will tell you more details.

Reply:
@shogohida Have a look at my [PR](https://github.com/pytorch/pytorch/pull/88807) 
Still don't understand, tag me. I will tell you more details.